src/test/scala/chiselTests/Mem.scala

```diff
@@ -188,6 +188,156 @@ private class TrueDualPortMemory(addrW: Int, dataW: Int) extends RawModule {
   }
 }
 
+class MemReadWriteTester extends BasicTester {
```

Comment:
I'm not a huge fan of big integration tests like this. However, this is entirely consistent with the rest of this existing file.

Can you open an issue to move all refactor this to move the integration tests into the integration test area and slim these down to just checking the CHIRRTL?